### PR TITLE
Create new [Gen 1] Random Auto Level Adjusted format

### PR DESCRIPTION
**Pull request by panpawn**
This format is very similar to the existing [Gen 1] Random format, except that all pokemon are initially at
level 50 and their levels are adjusted based on whether they win or lose in matches over time. So e.g. if
teams with Caterpie on them lose more often, Caterpie's level will be adjusted to be higher. As many matches are
played, each pokemon's level will eventually settle at a level so that it is as advantageous to have that
pokemon on a team as any other pokemon.
These match records are stored in the file config/pokemon-match-records.tsv - this records how many times each
pokemon has contributed to a win or loss, and its current adjusted level. This file does not exist initially -
it is created after the first battle in the new format, and updated after each subsequent battle. Its initial
state is stored in config/pokemon-match-records-initial.tsv. The code keeps track of the pokemon records by
passing around an object called pokemonRecord which is created by parsing the TSV file in room-battle.js.

Fix npm issue

### CODE REVIEW

**Comment by panpawn** [comment]
@Estu - so when you make changes, you'll want to push them to your branch called ``gen1-random-auto-level-adjusted``

**Comment by Estu** [comment]
Sounds good, thanks for the help.

**Comment by Estu** [comment]
No rush on any further review you need to do, but just in case this isn't clear, I don't think any further code changes are needed here.

**Comment by panpawn** [comment]
I thought you were going to make it auto generate the data file??

**Comment by Estu** [comment]
I implemented some very simple logic to read from an initial version of the data file if it doesn't already exist, and then create the actual data file the first time a battle ends.

**Comment by panpawn** [comment]
But the data file is still apart of this commit, which means everytime I update it will get overwritten with whatever is on the repository

**Comment by Estu** [comment]
The initial version of the data file is part of the commit - config/pokemon-match-records-initial.tsv. The actual data file - config/pokemon-match-records.tsv - is not.

File: config/formats.js [hunk 1]
```diff
@@ -802,6 +802,13 @@ exports.Formats = [
 		team: 'random',
 		ruleset: ['Pokemon', 'Standard'],
 	},
+	{
+		name: "[Gen 1] Random Auto Level Adjusted",
+
+		mod: 'gen1',
+		team: 'randomAutoLevelAdjusted',
+		ruleset: ['Pokemon', 'Standard'],
+	},
```
Comment (panpawn) [inline config/formats.js]: This should go under "Custom Gold Tiers"

**Comment by Estu** [comment]
Done - actually successfully squashed the new commit into the previous one, too, I think.

**Comment by panpawn** [comment]
Guess we'll see how this goes soon

**Comment by Estu** [comment]
Fantastic. When will I be able to see this on Gold?

**Comment by panpawn** [comment]
Next time I update and restart the server... maybe a few days?

**Comment by Estu** [comment]
Cool, I'll keep an eye out.

**Comment by Estu** [comment]
Thanks for all your help!

**Comment by panpawn** [comment]
I actually just patched this in - it should be live now!

**Comment by Estu** [comment]
Just played for an hour with my buddy - works great! He's interested in being able to see the match records as we play, so we can watch the level scaling evolve; if there's no easy way to do this, though, it's no big deal.

**Comment by panpawn** [comment]
I can make this a thing, sure

**Comment by panpawn** [comment]
Command is /randombattleadjusted https://github.com/panpawn/Gold-Server/commit/06749f6cda641d292bf415b5cf0c56f3e1779275

**Comment by Estu** [comment]
Awesome! Thanks!

**Comment by panpawn** [comment]
@Estu - I'm in the process of updating gold with main and I can't figure out how to get this working with main's changes.

Are you okay if I temporarily remove it, and then after I update, you can pull request the fix to make it work again?

**Comment by Estu** [comment]
Sure - I think I can look at it this weekend.

**Comment by panpawn** [comment]
@Estu thanks!

**Pull request merged**